server/service/service_global_schedule.go

```diff
@@ -0,0 +1,83 @@
+package service
+
+import (
+	"context"
+	"database/sql"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/pkg/errors"
+)
+
+func (svc *Service) GlobalScheduleQuery(ctx context.Context, sq *fleet.ScheduledQuery) (*fleet.ScheduledQuery, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+
+	gp, err := svc.ds.EnsureGlobalPack()
+	if err != nil {
+		return nil, err
+	}
+	sq.PackID = gp.ID
+
+	return svc.ScheduleQuery(ctx, sq)
+}
+
+func (svc *Service) GetGlobalScheduledQueries(ctx context.Context, opts fleet.ListOptions) ([]*fleet.ScheduledQuery, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+
+	gp, err := svc.ds.EnsureGlobalPack()
+	if err != nil {
+		return nil, err
+	}
+
+	return svc.ds.ListScheduledQueriesInPack(gp.ID, opts)
+}
+
+func (svc *Service) ModifyGlobalScheduledQueries(
+	ctx context.Context,
+	queryID uint,
+	query fleet.ScheduledQueryPayload,
+) (*fleet.ScheduledQuery, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Pack{}, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
+
+	gp, err := svc.ds.EnsureGlobalPack()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
What if `queryID` is the `scheduled_queries` ID, and then this method can call `svc.ModifyScheduledQuery` similar to what you've done on line 21?